src/main/java/org/traccar/geocoder/JsonGeocoder.java

```diff
@@ -88,7 +88,14 @@ public String getAddress(
             }
         }
 
-        Invocation.Builder request = Context.getClient().target(String.format(url, latitude, longitude)).request();
+        String urlFormat = String.format(url, latitude, longitude);
+        // The AutoNavi web API(V4.2.0)'s location Must Format "longitude,latitude".
+        boolean isAutoNavi = url.contains("restapi.amap.com/v3/geocode/regeo");
+        if (isAutoNavi){
+            urlFormat = String.format(url, longitude, latitude);
```

Comment:
What is this for?

Reply:
traccar's default “location” format is "latitude,longitude", but AutoNavi must be "longitude, latitude".
AutoNavi API Documentation like this:
```
传入内容规则：经度在前，纬度在后，经纬度间以“,”分割，经纬度小数点后不要超过 6 位。
```
translate:
```
Incoming content rules: longitude first, latitude last, and longitude and latitude are separated by ",", no more than 6 digits after the decimal point
```

Reply:
I see, but the solution is not generic. We would need something better.

Reply:
Add a  "UrlFormat" class like "AddressFormat"?

Reply:
I think we can either have placeholders like `{latitude}` or have a boolean flag. Formatter class is probably overkill here.

Reply:
ok. i will add a boolean flag.  do you hava a better key name?
```
    /**
     * Boolean flag to apply reverse location. 
     * Traccar's default locatin is "latitude,longitude", but some Geolocation Provider is "longitude, latitude".
     * Such as AutoNavi. Default value is false.
     */
    public static final ConfigKey GEOCODER_REQUEST_LOCATION_REVERSE= new ConfigKey(
            "geocoder.requestLocationReverse", Boolean.class);
```

Reply:
I think you misunderstood. It doesn't have to be configurable in the config file. Subclass should just provide this flag because it doesn't change.

Reply:
As for the name, I would prefer something more explicit like `latitudeFirst`.

Reply:
> I think you misunderstood. It doesn't have to be configurable in the config file. Subclass should just provide this flag because it doesn't change.

well. i think that would be a big change, involving too much.

Reply:
Why would it be a big change? Seems like a pretty trivial change to me.